neqo-http3/tests/priority.rs

```diff
@@ -0,0 +1,171 @@
+    assert!(out.as_dgram_ref().is_none());
+
+    let authentication_needed = |e| matches!(e, Http3ClientEvent::AuthenticationNeeded);
+    assert!(client.events().any(authentication_needed));
+    client.authenticated(AuthenticationStatus::Ok, now());
+
+    let out = client.process(out.dgram(), now());
+    let connected = |e| matches!(e, Http3ClientEvent::StateChange(Http3State::Connected));
+    assert!(client.events().any(connected));
+
+    assert_eq!(client.state(), Http3State::Connected);
+    // Exchange H3 setttings
+    let out = server.process(out.dgram(), now());
+    let out = client.process(out.dgram(), now());
+    let out = server.process(out.dgram(), now());
+    let out = client.process(out.dgram(), now());
+    let _ = server.process(out.dgram(), now());
+}
+
+fn connect() -> (Http3Client, Http3Server) {
+    let mut client = default_http3_client();
+    let mut server = default_http3_server();
+    connect_with(&mut client, &mut server);
+    (client, server)
+}
+
+#[test]
+fn priority_update() {
+    let (mut client, mut server) = connect();
+    let stream_id = client
+        .fetch(
+            Instant::now(),
+            "GET",
+            "https",
+            "something.com",
+            "/",
+            &[],
+            Priority::new(4, true),
+        )
+        .unwrap();
+    exchange_packets(&mut client, &mut server);
+
+    // get event of the above request, skipping events of the connection setup
+    let header_event = loop {
+        let event = server.next_event().unwrap();
+        if matches!(event, Http3ServerEvent::Headers { .. }) {
+            break event;
+        }
+    };
+
+    match header_event {
+        Http3ServerEvent::Headers {
+            request: _,
+            headers,
+            fin,
+        } => {
+            let expected_headers = vec![
+                Header::new(":method", "GET"),
+                Header::new(":scheme", "https"),
+                Header::new(":authority", "something.com"),
+                Header::new(":path", "/"),
+                Header::new("priority", "u=4,i"),
+            ];
+            assert_eq!(headers, expected_headers);
+            assert_eq!(fin, false);
+        }
+        other => panic!("unexpected server event: {:?}", other),
+    }
+
+    let update_priority = Priority::new(3, false);
+    client.priority_update(stream_id, update_priority).unwrap();
+    exchange_packets(&mut client, &mut server);
+
+    let priority_event = loop {
+        let event = server.next_event().unwrap();
+        if matches!(event, Http3ServerEvent::PriorityUpdate { .. }) {
+            break event;
+        }
+    };
+
+    match priority_event {
+        Http3ServerEvent::PriorityUpdate {
+            stream_id: update_id,
+            priority,
+        } => {
+            assert_eq!(update_id, stream_id);
+            assert_eq!(priority, update_priority);
+        }
+        other => panic!("unexpected server event: {:?}", other),
+    }
+}
```

Comment:
Please add test for the multiple priority update:
- the initial priority is 5 and priority_update() is called with value 5 -> there should not be any priority frame sent
- the initial priority is 5 and priority_update() is called with value 6 -> there should be a priority frame sent
- the initial priority is 5 and priority_update() is called with value 6 then once again with 5 -> there should not be any  priority frame sent
- the initial priority is 5 and priority_update() is called with value 6 then once again with 7 -> there should be a  priority frame sent with value 7
- the initial priority is 5 and priority_update() is called with value 6 and the stream is reset -> there should not be any  priority frame sent
- the initial priority is 5 and priority_update() is called with value 5 and a changed increment value  -> there should be a priority frame sent